content-src/lib/snippets.js

```diff
@@ -401,13 +400,24 @@ export function addSnippetsSubscriber(store) {
     ) {
       initializing = true;
       await snippets.init({appData: state.Snippets});
+      // istanbul ignore if
+      if (state.Prefs.values["asrouter.devtoolsEnabled"]) {
+        console.log("Legacy snippets initialized"); // eslint-disable-line no-console
+      }
       initializing = false;
     } else if (
-      (state.Prefs.values["feeds.snippets"] === false ||
-        state.Prefs.values.disableSnippets === true) &&
+      (
+        state.Prefs.values["feeds.snippets"] === false ||
+        state.Prefs.values.disableSnippets === true ||
```

Comment:
This is super complicated: As far as I know we control the availability of all our sections through the `feed.*` pref, when one of them is `false` the section is off. The SnippetsFeed only controls the legacy snippets implementation, how did we end up with a separate `disableSnippets` pref?
Glad this will go away.

Reply:
Since `state.Prefs.values["feeds.snippets"] ` is user-controlled, we needed a different pref that was non-user-accessible to turn off snippets for all users if some really bad event happened (i.e. `disableSnippets`)

I agree this code is extremely crappy and I am so excited for legacy onboarding and snippets to go away.